android/KMAPro/kMAPro/src/main/java/com/keyman/android/SystemKeyboard.java

```diff
@@ -64,11 +64,13 @@ public void onCreate() {
       KMManager.setDebugMode(true);
     }
     KMManager.addKeyboardEventListener(this);
-    KMManager.initialize(getApplicationContext(), KeyboardType.KEYBOARD_TYPE_SYSTEM);
-    interpreter = new KMHardwareKeyboardInterpreter(getApplicationContext(), KeyboardType.KEYBOARD_TYPE_SYSTEM);
+    Context context = getApplicationContext();
+    KMManager.initialize(context, KeyboardType.KEYBOARD_TYPE_SYSTEM);
+    DefaultLanguageResource.install(context);
```

Comment:
Will this reinstall sil_euro_latin if the user has deliberately removed it?

Reply:
DefaultLanguageReference line 34 stores a boolean if the default dictionary has ever been installed

```
editor.putBoolean(defaultKeyboardInstalled, true);
```
So future installs are skipped.

Reply:
>  default dictionary

or default keyboard?

Reply:
ah yeah
l 34 stores if default keyboard has ever been installed
l 58 stores if default dictionary has ever been installed